src/Hint/Lambda.hs

```diff
@@ -69,14 +71,18 @@ foo a b c = bar (flux ++ quux) c where flux = a -- foo a b = bar (flux ++ quux)
 foo a b c = bar (flux ++ quux) c where flux = c
 yes = foo (\x -> Just x) -- @Warning Just
 foo = bar (\x -> (x `f`)) -- f
-baz = bar (\x -> (x +)) -- (+)
+baz = bar (\x -> (x +)) -- (+) @NoRefactor
 foo = bar (\x -> case x of Y z -> z) -- \(Y z) -> z @NoRefactor
 yes = blah (\ x -> case x of A -> a; B -> b) -- \ case A -> a; B -> b @NoRefactor
+yes = blah (\ x -> case x of A -> a; B -> b) -- @Note may require `{-# LANGUAGE LambdaCase #-}` adding to the top of the file @NoRefactor
 no = blah (\ x -> case x of A -> a x; B -> b x)
+yes = blah (\ x -> (y, x)) -- (y,) @NoRefactor
 yes = blah (\ x -> (y, x, z+q)) -- (y, , z+q) @NoRefactor
+yes = blah (\ x -> (y, x, y, u, v)) -- (y, , y, u, v) @NoRefactor
+yes = blah (\ x -> (y, x, z+q)) -- @Note may require `{-# LANGUAGE TupleSections #-}` adding to the top of the file @NoRefactor
 yes = blah (\ x -> (y, x, z+x))
 tmp = map (\ x -> runST $ action x)
-yes = map (\f -> dataDir </> f) dataFiles -- (dataDir </>)
+yes = map (\f -> dataDir </> f) dataFiles -- (dataDir </>) @NoRefactor
```

Comment:
There is something weird going on here.

Even though I'm using `suggestN'` which adds an empty list of refactorings, this test still fails with "bad refactoring", because there is **some** refactoring suggested:
```
TEST FAILURE (BAD REFACTORING)
SRC: src/Hint/Lambda.hs:85:0
INPUT: yes = map (\f -> dataDir </> f) dataFiles
Refactor output is expected to contain: (dataDir </>)
Actual: yes = map (\f -> dataDir </> f) dataFiles
```

What's weirder to me is how even before my patch `suggestN` was still being used, yet this test didn't require `@NoRefactor`. There must be some magic that I'm missing..

Perhaps @zliu41 can clue me in?

Reply:
The test pass criteria is that refactoring output contains the suggested replacement (`ideaTo`). If the list of refactorings is empty, then refactoring does nothing and echoes the input, `yes = map (\f -> dataDir </> f) dataFiles`. It doesn't contain the suggested replacement, `(dataDir </>)`, so the test fails.

Previously the list of refactorings contains `(refact $ toSS o)` (https://github.com/ndmitchell/hlint/blob/master/src/Hint/Lambda.hs#L154), so it worked.